BeyondTrust/beyondtrust_modules/connector_pra_platform.py

```diff
@@ -0,0 +1,160 @@
+from sekoia_automation.connector import Connector, DefaultConnectorConfiguration
+
+from . import BeyondTrustModule
+from .client import ApiClient
+from .helpers import parse_session, parse_session_end_time, parse_session_list
+from .logging import get_logger
+from .metrics import EVENTS_LAG, FORWARD_EVENTS_DURATION, INCOMING_MESSAGES, OUTCOMING_EVENTS
+
+logger = get_logger()
+
+
+class BeyondTrustPRAPlatformConfiguration(DefaultConnectorConfiguration):
+    base_url: str = Field(..., description="Base URL")
+    client_id: str = Field(..., description="Client ID")
+    client_secret: str = Field(..., description="Client secret", secret=True)
+    frequency: int = Field(60, description="Batch frequency in seconds")
+
+
+class BeyondTrustPRAPlatformConnector(Connector):
+    module: BeyondTrustModule
+    configuration: BeyondTrustPRAPlatformConfiguration
+
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+
+        self.cursor = CheckpointTimestamp(
+            time_unit=TimeUnit.SECOND,
+            path=self._data_path,
+            start_at=timedelta(days=1),
+            ignore_older_than=timedelta(days=7),
+        )
+        self.from_date = self.cursor.offset
+
+    @cached_property
+    def client(self) -> ApiClient:
+        return ApiClient(
+            base_url=self.configuration.base_url,
+            client_id=self.configuration.client_id,
+            client_secret=self.configuration.client_secret,
+        )
+
+    def _handle_response_error(self, response: requests.Response):
+        if not response.ok:
+            level = "critical" if response.status_code in [401, 403] else "error"
+
+            message = f"Request to BeyondTrust API failed with status {response.status_code} - {response.reason}"
+
+            # enrich error logs with detail from the Okta API
+            try:
+                error = response.json()
+                logger.error(
+                    message,
+                    error_message=error.get("message"),
+                    error_number=error.get("number"),
+                )
+
+            except Exception as e:
+                pass
+
+            self.log(message=message, level=level)
+
+    def fetch_events(self) -> Generator[list, None, None]:
+        # 1. Get list of sessions
+        # 2. Download details for every session
+        most_recent_date_seen = self.from_date
+
+        response = self.client.post(
+            f"{self.configuration.base_url}/api/reporting",
+            data={"generate_report": "AccessSessionListing", "duration": 0, "end_time": most_recent_date_seen},
+            headers={"Content-Type": "application/x-www-form-urlencoded"},
+            timeout=60,
+        )
+        self._handle_response_error(response)
+
+        if "<error>" in response.text and response.status_code == 200:
+            # no sessions
+            self.log(response.text, level="debug")
```

Comment:
As an error, I think that the `error` level is more suitable here
```suggestion
            self.log(response.text, level="error")
```

Reply:
It's not an error per se, they're just saying there's no events
```
<error >No Support report information matching your chosen criteria is available.</error>
```